docs/pages/api/rbac.mdx

```diff
@@ -0,0 +1,1017 @@
+	initTimeout              = time.Duration(30) * time.Second
+	identityFilePath  string = "auth.pem"
+	kubeconfigPath    string = "kubeconfig"
+	clusterName       string = "minikube"
+	roleAnnotationKey string = "create-teleport-role"
+)
+```
+
+We will use these constants later in the program. They define some values we may
+want to change later, including:
+
+|Constant|Description|
+|---|---|
+|`proxyAddr`|The host and port of the Teleport Proxy Service, e.g., `mytenant.teleport.sh:443`, which we will use to connect the client to your cluster. **Assign this to your own Proxy Service's host and port.**|
+|`initTimeout`|The timeout for connecting to the Teleport cluster. We have defined this as 30 seconds.|
+|`identityFilePath`|The path to the Teleport identity file you created earlier.|
+|`clusterName`|The name of the Kubernetes cluster you will fetch RBAC resources from. In this guide, the cluster's name is `minikube`.|
+|`roleAnnotationKey`|In Kubernetes, annotations are arbitrary key/value pairs that you can add to resources. The role and cluster role bindings we created earlier have the annotation key we specify here so our client application can fetch them.|
+
+### Initialize a Kubernetes RBAC client
+
+To contact the Kubernetes API, we will need to set up an HTTP client. The client
+authenticates to the API using mutual TLS, loading a client certificate,
+certificate authority, and private key from the file at `kubeconfigPath`.
+Earlier in the guide, we requested this from the Teleport Auth Service.
+
+Add the following below the `const` block:
+
+```go
+func getRBACClient() (v1.RbacV1Interface, error) {
+	f, err := os.Open(kubeconfigPath)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	kc, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	n, err := clientcmd.RESTConfigFromKubeConfig(kc)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	c, err := kubernetes.NewForConfig(n)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return c.RbacV1(), nil
+}
+```
+
+`getRBBACClient` opens and reads the Kubernetes credentials file at
+`kubeconfigPath`, then uses the file to set up a Kubernetes API client
+configuration (`clientcmd.RESTConfigFromKubeConfig(kc)`) and, with that, an HTTP
+client (`kubernetes.NewForConfig(n)`). 
+
+Finally, it returns an interface to the Kubernetes API dedicated to role-based
+access controls, which the rest of the program uses to interact with your
+Kubernetes cluster.
+
+### Create a Teleport role from a Kubernetes cluster role binding
+
+Add the following function below `getRBACClient`. This creates a Teleport role
+from a Kubernetes cluster role binding by populating fields in the former based
+on fields in latter:
+
+```go
+func createTeleportRoleFromClusterRoleBinding(teleport *client.Client, k types.KubeCluster, r rbacv1.ClusterRoleBinding) error {
+	if e, ok := r.Annotations[roleAnnotationKey]; !ok || e != "true" {
+		return nil
+	}
+
+	role := types.RoleV6{}
```

Comment:
Do you think users might find it confusing that the roles we asked them to create at the beginning are v5 roles and now we're using the API to create v6 roles?

Reply:
That was an oversight on my part—I think I was copying an example role from another guide that used v5 without thinking about it.